cmake-proxies/cmake-modules/AudacityTesting.cmake

```diff
@@ -0,0 +1,143 @@
+if( ${_OPT}has_tests )
+   if( CMAKE_CONFIGURATION_TYPES )
+      set( TESTS_DIR "${CMAKE_BINARY_DIR}/tests" )
+      set( TESTS_DEST_DIR "${CMAKE_BINARY_DIR}/tests/${CMAKE_CFG_INTDIR}" )
+   else()
+      set( TESTS_DIR "${CMAKE_BINARY_DIR}/tests" )
+      set( TESTS_DEST_DIR "${CMAKE_BINARY_DIR}/tests/${CMAKE_BUILD_TYPE}" )
+   endif()
+
+   # Install Catch2 framework
+
+   add_conan_lib(
+      Catch2
+      catch2/2.13.8
+      REQUIRED
+      ALWAYS_ALLOW_CONAN_FALLBACK
+   )
+
+   # Setup default CTest arguments when running from IDE
+
+   set(CMAKE_CTEST_ARGUMENTS "--output-on-failure;--verbose;${CMAKE_CTEST_ARGUMENTS}")
+
+   enable_testing()
+
+   #[[
+      add_unit_test(NAME name SOURCES file1 ... LIBRARIES lib1 ...)
+
+      Creates an executable called ${name}-test from the sources files ${file1}, ... and linked
```

Comment:
"source files"

Reply:
It may be useful to add `PROPERTIES prop1 ...` and pass those to: https://cmake.org/cmake/help/latest/command/set_tests_properties.html?highlight=set_tests_properties

Reply:
And here are all the possible properties https://cmake.org/cmake/help/latest/manual/cmake-properties.7.html#test-properties

Reply:
Maybe the function will have other keyword arguments to abbreviate other combinations of properties.  Don't ask me yet, what exactly.  Experience will tell what is useful.